### PR TITLE
Consolidate language tutorials

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
Consolidates quickstarts into one with a tabbed interface for different languages.

### CODE REVIEW

File: docs/reference/_static/expand_tabs.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Expands a specific tab of sphinx-tabs.
```
[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply docs/reference/_static/expand_tabs.js:2]: Hmm, most of this code came from elsewhere, can we really © it?

[the remark this answers is not in the document]
Comment (ChrisChinchilla) [reply docs/reference/_static/expand_tabs.js:2]: OK, I changed quite a bit, but I will add something.

File: docs/reference/src/quickstart/getting_started.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+:::
+
+Also in `CMakeLists.txt` the line
+`slint_target_sources(my_application ui/appwindow.slint)` is a Slint function used to
+add the `appwindow.slint` file to the target.
+
+Change the contents of `ui/appwindow.slint` to the following:
+
+:::{literalinclude} appwindow.slint
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/getting_started.md:44]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+:::{literalinclude} main_initial.js
+:lines: 6-10
+:::
+
+The `slint.loadFile` method resolves files from the process's current working directory, so from the `package.json` file's location.
+
+Replace the contents of `ui/appwindow.slint` with the following:
+
+:::{literalinclude} memory.slint
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/getting_started.md:121]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/memory_tile.md [hunk 4]
```diff
@@ -12,17 +12,65 @@ Lengths in Slint have a unit, here, the `px` suffix.
...
+:sync: rust
+
+Copy the following code inside of the `slint!` macro, replacing the current content:
+
+:::
+
+::::
+
+:::{literalinclude} memory_tile.slint
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/memory_tile.md:41]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/from_one_to_multiple_tiles.md [hunk 5]
```diff
@@ -19,17 +19,63 @@ The <span class="hljs-keyword">for</span> loop is declarative and automatically
...
+:sync: rust
+
+First, add the tile data structure definition at the top of the `slint!` macro:
+
+:::
+
+::::
+
+:::{literalinclude} main_multiple_tiles.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/from_one_to_multiple_tiles.md:47]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/polishing_the_tile.md [hunk 6]
```diff
@@ -26,9 +26,10 @@ _solved_ property used to animate the color to a shade of green when a player fi
 
 Replace the code inside the `ui/appwindow.slint` file with the following:
 
-```slint
-{{#include ../../rust/src/main_polishing_the_tile.rs:tile}}
-```
+:::{literalinclude} main_polishing_the_tile.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/polishing_the_tile.md:30]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/from_one_to_multiple_tiles.md [hunk 7]
```diff
@@ -19,17 +19,63 @@ The <span class="hljs-keyword">for</span> loop is declarative and automatically
...
+:sync: rust
+
+Next, replace the _export component <span class="hljs-title">MainWindow</span> inherits Window { ... }_ section at the bottom of the `slint!` macro with the following:
+
+:::
+
+::::
+
+:::{literalinclude} main_multiple_tiles.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/from_one_to_multiple_tiles.md:76]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/game_logic.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+the location of the icons, and then closes the curtains again.
+
+:::::{tab-set}
+::::{tab-item} C++
+:sync: cpp
+
+Add the following code inside the <span class="hljs-title">MainWindow</span> component to signal to the C++ code when the user clicks on a tile.
+
+:::{literalinclude} main_game_logic_in_rust.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/game_logic.md:21]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/game_logic.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+This change adds a way for the <span class="hljs-title">MainWindow</span> to call to the C++ code that it should
+check if a player has solved a pair of tiles. The Rust code needs an additional property to toggle to disable further
+tile interaction, to prevent the player from opening more tiles than allowed. No cheating allowed!
+
+The last change to the code is to act when the <span class="hljs-title">MemoryTile</span> signals that a player clicked it.
+
+Add the following handler in the <span class="hljs-title">MainWindow</span> `for` loop `clicked` handler:
+
+:::{literalinclude} main_game_logic_in_rust.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/game_logic.md:34]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/game_logic.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+::::
+
+::::{tab-item} NodeJS
+:sync: nodejs
+
+Change the contents of `memory.slint` to signal to the JavaScript code when the user clicks on a tile.
+
+:::{literalinclude} main_game_logic_in_rust.rs
+:language: slint
```
Comment (tronical) [inline docs/reference/src/quickstart/game_logic.md:63]:
```suggestion
:language: slint,no-preview
```

File: docs/reference/src/quickstart/getting_started.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+::::
+
+::::{tab-item} Rust
+:sync: rust
+
+We recommend using [rust-analyzer](https://rust-analyzer.github.io) and [our editor integrations for Slint](https://github.com/slint-ui/slint/tree/master/editors) for following this tutorial.
+
+Slint has an application template you can use to create a project with dependencies already set up that follows recommended best practices.
+
+Before using the template, install `[cargo-generate](https://github.com/cargo-generate/cargo-generate)`:
```
Comment (tronical) [inline docs/reference/src/quickstart/getting_started.md:138]:
```suggestion
Before using the template, install [`cargo-generate`](https://github.com/cargo-generate/cargo-generate):
```

File: docs/reference/src/quickstart/from_one_to_multiple_tiles.md [hunk 12]
```diff
@@ -19,17 +19,63 @@ The <span class="hljs-keyword">for</span> loop is declarative and automatically
...
 
-```slint
-{{#include ../../rust/src/main_multiple_tiles.rs:tile_data}}
-```
+:::
+
+:::{tab-item} NodeJS
+:sync: nodejs
+
+First, add the tile data structure definition at the top of the `memory.slint` file:
```
Comment (tronical) [inline docs/reference/src/quickstart/from_one_to_multiple_tiles.md:33]:
```suggestion
First, add the tile data structure definition at the top of the `ui/appwindow.slint` file:
```

File: docs/reference/src/quickstart/from_one_to_multiple_tiles.md [hunk 13]
```diff
@@ -19,17 +19,63 @@ The <span class="hljs-keyword">for</span> loop is declarative and automatically
...
 
-```slint
-{{#include ../../rust/src/main_multiple_tiles.rs:main_window}}
-```
+:::
+
+:::{tab-item} NodeJS
+:sync: nodejs
+
+Next, replace the _export component <span class="hljs-title">MainWindow</span> inherits Window { ... }_ section at the bottom of the `memory.slint` file with the following:
```
Comment (tronical) [inline docs/reference/src/quickstart/from_one_to_multiple_tiles.md:62]:
```suggestion
Next, replace the _export component <span class="hljs-title">MainWindow</span> inherits Window { ... }_ section at the bottom of the `ui/appwindow.slint` file with the following:
```

File: docs/reference/src/quickstart/creating_the_tiles.md [hunk 14]
```diff
@@ -1,20 +1,66 @@
...
+Slint generates a getter and a setter function. In this case `get_memory_tiles` and `set_memory_tiles`.
+Since `memory_tiles` is a Slint array represented as a [`Rc<dyn slint::Model>`](https://slint.dev/docs/rust/slint/trait.Model).
+
+::::
+
+:::::
+
 You can't change the model generated by Slint, but you can extract the tiles from it and put them
 in a [`slint::VectorModel`](https://slint.dev/docs/cpp/api/classslint_1_1vectormodel) which inherits from `Model`.
 `VectorModel` lets you make changes and you can use it to replace the static generated model.
```
Comment (tronical) [inline docs/reference/src/quickstart/creating_the_tiles.md:66]: This paragraph is language specific, it belongs into the C++ section. There are different sections in `creating_the_tiles_from_js.md` and `creating_the_tiles_from_rust.md`

**Review by tronical — APPROVED** [review]

**Pull request merged**